sizefashion/app/components/Carousel/index.tsx

```diff
@@ -0,0 +1,97 @@
+"use client";
+import { useItemsContext } from '@context/ItemsContext';
+import React from 'react';
+import { FaChevronLeft, FaChevronRight } from 'react-icons/fa';
+
+const scrollAmount = 720;
+
+const Carousel = (): React.JSX.Element => {
+
+  const ItemsContext = useItemsContext();
+
+  React.useEffect(() => {
+    let scrollableDiv: HTMLElement | null;
+    let scrollLeftButton: HTMLElement | null;
+    let scrollRightButton: HTMLElement | null;
+
+    if (document) {
+      scrollableDiv = document.querySelector('.scrollable-div');
+      scrollLeftButton = document.getElementById('scrollLeftButton');
+      scrollRightButton = document.getElementById('scrollRightButton');
+
+      if (scrollableDiv) {
+
+        if (scrollLeftButton) {
+          scrollLeftButton.addEventListener('click', () => {
+            scrollableDiv?.scrollBy({
+              left: -scrollAmount,
+              behavior: 'smooth',
+            });
+          });
+
+          scrollLeftButton?.addEventListener('mouseenter', () => handleFocus(scrollLeftButton));
+          scrollLeftButton?.addEventListener('mouseleave', () => handleBlur(scrollLeftButton));
+        };
+
+        if (scrollRightButton) {
+          scrollRightButton.addEventListener('click', () => {
+            scrollableDiv?.scrollBy({
+              left: scrollAmount,
+              behavior: 'smooth',
+            });
+          });
+
+          scrollRightButton?.addEventListener('mouseenter', () => { handleFocus(scrollRightButton) });
+          scrollRightButton?.addEventListener('mouseleave', () => { handleBlur(scrollRightButton) });
+        };
+      };
+    };
+
+    return () => {
+      if (document) {
+        if (scrollableDiv) scrollableDiv = null;
+        if (scrollLeftButton) scrollLeftButton.removeEventListener('focus', () => { })
+        if (scrollRightButton) scrollRightButton.removeEventListener('focus', () => { })
+      };
+    };
+
+  }, []);
+
+  const handleFocus = (element: HTMLElement | null): void => {
+    if (element) element.style.backgroundColor = '#e2f3ff'
+  };
+
+  const handleBlur = (element: HTMLElement | null): void => {
+    if (element) element.style.backgroundColor = '#d1ecff'
+  };
+
+  return (
+    <div className='flex w-full'>
+
+      <div id="scrollLeftButton" className='flex items-center justify-center cursor-pointer w-16 text-white transition select-none absolute h-80 left-0'>
+        <FaChevronLeft />
+      </div>
+
+      <div className='bg-blue-200 flex w-full px-16'>
+        <div className='scrollable-div flex flex-row overflow-x-hidden overflow-y-hidden whitespace-nowrap'>
+          {ItemsContext.dataFromSource.map((item, key) =>
+            <div
+              className='flex h-80 object-cover'
+              style={{ backgroundColor: 'white', minWidth: 360 }}
+              key={key}
+            >
+              {/* YOLO */}<div dangerouslySetInnerHTML={{ __html: item.image }} />
```

Comment:
@vaporwavie given that the payload appears as shown in the screenshot below, I opted for two different rendering approaches. However, there might be a more efficient way to handle this.

![image](https://github.com/sizebay/frontend-challenge/assets/99999956/e7621ac8-bee2-4f5b-bd2f-12188563f7e4)

Reply:
@mbdesousa here's a cool solution to adopt with React, in two steps:

### 1. Sanitizing the string element
```js
// Given this dirty string
let dirtyProductImage =
 "<img src=\"//store.sizebay.com/cdn/shop/products/foto-05_large.png?v=1601062306\" alt=\"Nylon Bomber Jacket in Stone\" class=\"product__img\">";

// We first remove the backlashes
dirtyProductImage = dirtyProductImage.replace(/\"/g, `"`);

// Then we append the missing closing tag to prevent breaking layouts (while also increasing a11y coverage)
const storeLogo = `${dirtyProductImage}</img>`;
```

### 2. Rendering it
```jsx
function ProductImage({ productImage }) {
  const imageSlotRef = useRef(null);

  useEffect(() => {
    if (!imageSlotRef.current) return;

    // We're going to replace the image element with the one brought from props
    imageSlotRef.current.innerHTML = productImage;
  }, [imageSlotRef]);

  return (
    <div style={{ display: "contents" }} ref={imageSlotRef}>
      {/* We're going to replace this image element with the one brought from props */}
    </div>
  );
}

```

Using `dangerouslySetInnerHTML` can expose your application to XSS atacks. See at: https://legacy.reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml